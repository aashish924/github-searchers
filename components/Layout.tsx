import React, { ReactNode } from 'react'
import Link from 'next/link'
import Head from 'next/head'

import styled from 'styled-components';


type Props = {
  children?: ReactNode
  title?: string
}


const Container = styled.div`
    margin: 48px;
`;

const Layout = ({ children, title = 'This is the default title' }: Props) => (
  <div>
    <Head>
      <title>{title}</title>
      <meta charSet="utf-8" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
      <Container>
        {children}
      </Container>
  </div>
);

export default Layout
