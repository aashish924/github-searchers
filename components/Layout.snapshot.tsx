import * as React from 'react';
import renderer from 'react-test-renderer';

import Layout from './Layout'


describe('Layout should render correctly', () => {
  it('Layout without children', () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot('layout-without-children');
  });

  it('Layout with custom title', () => {
    const tree = renderer.create(<Layout title="Github Searchers" />).toJSON();
    expect(tree).toMatchSnapshot('layout-with-custom-title');
  });

  it('Layout with some children', () => {
    const page = <Layout title="Github Searchers">
      <header>Github Searchers</header>
      <section className="main">
        <span>This is main area</span>
      </section>
      <section className="footer">
        <span>This is footer</span>
      </section>
    </Layout>;
    const tree = renderer.create(page).toJSON();
    expect(tree).toMatchSnapshot('layout-with-children');
  });
});