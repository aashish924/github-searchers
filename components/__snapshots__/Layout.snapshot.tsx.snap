// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly Layout with custom title: layout-with-custom-title 1`] = `
<div>
  <div
    className="sc-AxjAm iBuasp"
  />
</div>
`;

exports[`Layout should render correctly Layout with some children: layout-with-children 1`] = `
<div>
  <div
    className="sc-AxjAm iBuasp"
  >
    <header>
      Github Searchers
    </header>
    <section
      className="main"
    >
      <span>
        This is main area
      </span>
    </section>
    <section
      className="footer"
    >
      <span>
        This is footer
      </span>
    </section>
  </div>
</div>
`;

exports[`Layout should render correctly Layout without children: layout-without-children 1`] = `
<div>
  <div
    className="sc-AxjAm iBuasp"
  />
</div>
`;
